chainer/functions/loss/contrastive.py

```diff
@@ -0,0 +1,90 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class Contrastive(function.Function):
+
+    """Contrastive loss function."""
+
+    def __init__(self, margin):
+        self.margin = float(margin)
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 3)
+
+        x0_type, x1_type, y_type = in_types
+        type_check.expect(
+            x0_type.dtype == numpy.float32,
+            x1_type.dtype == numpy.float32,
+            x0_type.shape == x1_type.shape,
+            x0_type.shape[0] == x1_type.shape[0],
```

Comment:
This equality is redundant because we have already checked it in the previous line.